pyrate/core/phase_closure/sum_closure.py

```diff
@@ -0,0 +1,140 @@
+        _ifg.close()
+
+    ret_combined = join_dicts(mpiops.comm.allgather(ret_combined))
+    return ret_combined
+
+
+def sum_phase_closures(ifg_files: List[str], loops: List[WeightedLoop], params: dict) -> \
+        Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    edge_to_indexed_ifgs = __create_ifg_edge_dict(ifg_files, params)
+    ifgs = [v.IfgPhase for v in edge_to_indexed_ifgs.values()]
+    n_ifgs = len(ifgs)
+    ifg0 = ifgs[0]
+
+    closure_dict = {}
+    check_ps_dict = {}
+    if params[cf.PARALLEL]:
+        # rets = Parallel(n_jobs=params[cf.PROCESSES], verbose=joblib_log_level(cf.LOG_LEVEL))(
+        #     delayed(__compute_check_ps)(ifg0, n_ifgs, weighted_loop, edge_to_indexed_ifgs, params)
+        #     for weighted_loop in loops
+        # )
+        # for k, r in enumerate(rets):
+        #     closure_dict[k], check_ps_dict[k] = r
+        # TODO: enable multiprocessing - needs pickle error fix
+        for k, weighted_loop in enumerate(loops):
+            closure_dict[k], check_ps_dict[k] = __compute_check_ps(
+                ifg0, n_ifgs, weighted_loop, edge_to_indexed_ifgs, params
+            )
+    else:
+        loops_with_index = list(enumerate(loops))
+        process_loops = mpiops.array_split(loops_with_index)
+        for k, weighted_loop in process_loops:
+            closure_dict[k], check_ps_dict[k] = __compute_check_ps(
+                ifg0, n_ifgs, weighted_loop, edge_to_indexed_ifgs, params
+            )
+        closure_dict = join_dicts(mpiops.comm.gather(closure_dict, root=0))
+        check_ps_dict = join_dicts(mpiops.comm.gather(check_ps_dict, root=0))
+
+    closure, check_ps, num_occurences_each_ifg = None, None, None
+    if mpiops.rank == 0:
+        num_occurences_each_ifg = _find_num_occurences_each_ifg(loops, edge_to_indexed_ifgs, n_ifgs)
+        closure = np.dstack([v for k, v in sorted(closure_dict.items(), key=lambda x: x[0])])
+        check_ps = np.sum(np.stack([v for k, v in sorted(check_ps_dict.items(), key=lambda x: x[0])], axis=3), axis=3)
+
+    return closure, check_ps, num_occurences_each_ifg
+
+
+def _find_num_occurences_each_ifg(loops, edge_to_indexed_ifgs, n_ifgs):
+    """find how many times each ifg appears in total in all loops"""
+    num_occurences_each_ifg = np.zeros(shape=n_ifgs, dtype=np.uint16)
+    for weighted_loop in loops:
+        for signed_edge in weighted_loop.loop:
+            indexed_ifg = edge_to_indexed_ifgs[signed_edge.edge]
+            ifg_index = indexed_ifg.index
+            num_occurences_each_ifg[ifg_index] += 1
+    return num_occurences_each_ifg
+
+
+def __compute_check_ps(ifg: Ifg, n_ifgs: int, weighted_loop: WeightedLoop,
+                       edge_to_indexed_ifgs: Dict[Edge, IndexedIfg], params: dict) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    find sum `closure` of each loop, and compute `check_ps` for each pixel.
+    PS: Persistent Scatterer
+    """
+    large_dev_thr = params[cf.LARGE_DEV_THR]
```

Comment:
My understanding is `large_dev_thr` is given as a number in radians, e.g. 1.5078 is the current default.
I think it would be easier for users to define this as a multiplier of pi radians. For instance, instead of giving pi/2 as 1.5708 in the config file, a user would specify 0.5. Here in the code, that number would be multiplied by pi, to give the number 1.5708 actually used for thresholding.

Reply:
addressed by https://github.com/GeoscienceAustralia/PyRate/pull/307/commits/d6872fcf97e19722639c07bb4029d7b16fba68d8